tests/services/test_scheduler.py

```
@@ -97,24 +98,199 @@ def test_decrypt_location_job(self, db_session, storage_location, file_inventory
+        assert loc.permissions_error is None
+        assert mp.permissions_error is None
+
+    def test_dispatches_notification_on_first_error(self, db_session, tmp_path, monkeypatch):
+        """A STORAGE_PERMISSION_ERROR notification is dispatched the first time an error is detected."""
+        from app.services.scheduler import check_storage_permissions_job_func
+
+        cold_path = tmp_path / "cold"
+        cold_path.mkdir()
+        loc = self._make_cold_location(db_session, str(cold_path))
+
+        monkeypatch.setattr("app.services.scheduler.os.access", lambda path, mode: mode != 2)
```

Ruff (0.15.9)

[warning] Unused lambda argument: `path`

(ARG005)